netbox/netbox/api/serializers/base.py

```diff
@@ -36,6 +56,15 @@ def __init__(self, *args, nested=False, fields=None, **kwargs):
         if self.nested and not fields:
             self._requested_fields = getattr(self.Meta, 'brief_fields', None)
 
+        # don't override the field if the class already defines these so can set lookup_field
+        if ("url" in self.fields and not isinstance(self.fields["url"], serializers.HyperlinkedIdentityField) and
+                isinstance(self.fields["url"], serializers.RelatedField)):
+            self.fields["url"] = NetBoxApiHyperlinkedIdentityField(self.Meta.model)
+        if ("display_url" in self.fields and not
+                isinstance(self.fields["display_url"], serializers.HyperlinkedIdentityField) and
+                isinstance(self.fields["display_url"], serializers.RelatedField)):
+            self.fields["display_url"] = NetBoxUrlHyperlinkedIdentityField(self.Meta.model)
+
```

Comment:
Instead of this, we should be able to override the `get_url()` method on our both of our HyperlinkedIdentityField subclasses to infer the model from the serializer (`self.model.parent`) automatically. (We can pass a dummy value to `view_name` to avoid the assertion checks.)